solidity/contracts/relay/RelayMaintainerProxy.sol

```diff
@@ -0,0 +1,100 @@
+/// @notice The proxy contract that allows the relay maintainers to be refunded
+///         for the spent gas from the `ReimbursementPool`. When proving the
+///         next Bitcoin difficulty epoch, the maintainer calls the
+///         `LightRelayMaintainerProxy` which in turn calls the actual `LightRelay`
+///         contract.
+contract LightRelayMaintainerProxy is Ownable, Reimbursable {
+    ILightRelay public lightRelay;
+
+    /// @notice Stores addresses that can maintain the relay.
+    mapping(address => bool) public relayMaintainers;
+
+    /// @notice Gas that is meant to balance the retarget overall cost. Can be
+    //          updated by the governance based on the current market conditions.
+    uint256 public retargetGasOffset;
+
+    event RelayUpdated(address newRelay);
+
+    event RelayMaintainerStatusChanged(
+        address indexed maintainer,
+        bool isAuthorized
+    );
+
+    modifier onlyRelayMaintainer() {
+        require(relayMaintainers[msg.sender], "Caller is not authorized");
+        _;
+    }
+
+    constructor(ILightRelay _lightRelay, ReimbursementPool _reimbursementPool) {
+        lightRelay = _lightRelay;
+        reimbursementPool = _reimbursementPool;
+
+        // TODO: Set the proper value or remove completely.
+        retargetGasOffset = 0;
+    }
+
+    /// @notice Wraps `LightRelay.retarget` call and reimburses the caller's
+    ///         transaction cost. Can only be called by an authorized relay
+    ///         maintainer.
+    /// @dev See `LightRelay.retarget` function documentation.
+    function retarget(bytes memory headers) external onlyRelayMaintainer {
+        uint256 gasStart = gasleft();
+
+        lightRelay.retarget(headers);
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + retargetGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Allows the governance to upgrade the `LightRelay` address.
+    /// @dev The function does not implement any governance delay and does not
+    ///      check the status of the `LightRelay`. The Governance implementation
+    ///      needs to ensure all requirements for the upgrade are satisfied
+    ///      before executing this function.
+    function updateLightRelay(ILightRelay _lightRelay) external onlyOwner {
+        lightRelay = _lightRelay;
+
+        emit RelayUpdated(address(_lightRelay));
+    }
+
+    /// @notice Sets authorization status from the given address. Can only be
+    ///         called by the owner.
+    /// @param maintainer The address of the maintainer to be authorized or
+    ///                   deauthorized.
+    function setRelayMaintainerStatus(address maintainer, bool isAuthorized)
+        external
+        onlyOwner
+    {
+        relayMaintainers[maintainer] = isAuthorized;
+        emit RelayMaintainerStatusChanged(maintainer, isAuthorized);
+    }
+
+    // TODO: Most likely `onlyReimbursableAdmin()` needs to be overridden.
```

Comment:
Yes, see how it's done in `MaintainerProxy`:

https://github.com/keep-network/tbtc-v2/blob/4523b83f2b4612d93d91ab55e254bb13cec632a1/solidity/contracts/maintainer/MaintainerProxy.sol#L154-L157

Looking at `Reimbursable` from another perspective, we could use the same trick OpenZeppelin uses: expose `Reimbursable__init()` that would set `msg.sender` as reimbursable admin. But the current version is good enough, so let's leave it as is, especially since it is used in already deployed contracts.

Reply:
Done in c41b25a3041d653c929742b0c37859fd16d2bfab.
Used the same version as in `MaintainerProxy`:
```
    modifier onlyReimbursableAdmin() override {
        require(owner() == msg.sender, "Caller is not the owner");
        _;
    }
```